routes/cube_routes.js

```diff
@@ -970,6 +896,48 @@ router.get('/samplepackimage/:id/:seed', async (req, res) => {
   }
 });
 
+async function updateCubeAndBlog(req, res, cube, changelog, added, missing) {
+  const blogpost = new Blog();
+  blogpost.title = 'Cube Bulk Import - Automatic Post';
+  blogpost.html = changelog;
+  blogpost.owner = cube.owner;
+  blogpost.date = Date.now();
+  blogpost.cube = cube._id;
+  blogpost.dev = 'false';
+  blogpost.date_formatted = blogpost.date.toLocaleString('en-US');
+  blogpost.username = cube.owner_name;
+  blogpost.cubename = cube.name;
+
+  if (missing.length > 0) {
+    const reactProps = {
+      cubeID: req.params.id,
+      missing,
+      added: added.map(({ _id, name, image_normal, image_flip }) => ({ _id, name, image_normal, image_flip })),
+      blogpost: blogpost.toObject(),
+    };
+    return res.render('cube/bulk_upload', {
+      reactHTML: BulkUploadPage
+        ? ReactDOMServer.renderToString(React.createElement(BulkUploadPage, reactProps))
+        : undefined,
+      reactProps: serialize(reactProps),
+      cube,
+      cube_id: req.params.id,
+      title: `${abbreviate(cube.name)} - Bulk Upload`,
+    });
+  }
+  await blogpost.save();
+  cube = setCubeType(cube, carddb);
+  try {
```

Comment:
we should wrap this entire function in a try/catch.

Reply:
This should always be called from a route that's wrapped in a try catch block.

Reply:
Doesn't matter, it costs nothing, and can help prevent issues in the future

Reply:
Would like to see this change still